xarray/convert.py

```diff
@@ -53,3 +65,153 @@ def set_cdms2_attrs(var, attrs):
+
+    # Create the right type of masked array (should be easier after #1769)
+    if isinstance(dataarray.data, Array):
+        masked_data = dask_ma.masked_invalid(dataarray)
+    else:
+        masked_data = np.ma.masked_invalid(dataarray)
+
+    cube = iris.cube.Cube(masked_data, **args)
+
+    return cube
+
+
+def _iris_obj_to_attrs(obj):
+    """ Return a dictionary of attrs when given a Iris object
+    """
+    attrs = {'standard_name': obj.standard_name,
+             'long_name': obj.long_name}
+    if obj.units.calendar:
+        attrs['calendar'] = obj.units.calendar
+    if obj.units.origin != '1':
+        attrs['units'] = obj.units.origin
+    attrs.update(obj.attributes)
+    return dict((k, v) for k, v in attrs.items() if v is not None)
+
+
+def _iris_cell_methods_to_str(cell_methods_obj):
+    """ Converts a Iris cell methods into a string
+    """
+    cell_methods = []
+    for cell_method in cell_methods_obj:
+        names = ''.join(['{}: '.format(n) for n in cell_method.coord_names])
+        intervals = ' '.join(['interval: {}'.format(interval)
+                              for interval in cell_method.intervals])
+        comments = ' '.join(['comment: {}'.format(comment)
+                             for comment in cell_method.comments])
+        extra = ' '.join([intervals, comments]).strip()
+        if extra:
+            extra = ' ({})'.format(extra)
+        cell_methods.append(names + cell_method.method + extra)
+    return ' '.join(cell_methods)
+
+
+def from_iris(cube):
+    """ Convert a Iris cube into an DataArray
+    """
+    import iris.exceptions
+    try:
+        from dask.array import ma as dask_ma
+        from dask.array import Array
+    except ImportError:
+        dask_ma = None
+
+    name = cube.var_name
+    dims = []
+    for i in range(cube.ndim):
+        try:
+            dim_coord = cube.coord(dim_coords=True, dimensions=(i,))
+            dims.append(dim_coord.var_name)
+        except iris.exceptions.CoordinateNotFoundError:
+            dims.append("dim_{}".format(i))
+
+    coords = OrderedDict()
+
+    for coord in cube.coords():
+        coord_attrs = _iris_obj_to_attrs(coord)
+        coord_dims = [dims[i] for i in cube.coord_dims(coord)]
+        if not coord.var_name:
+            raise ValueError("Coordinate '{}' has no var_name attribute".format(coord.name()))
+        if coord_dims:
+            coords[coord.var_name] = (coord_dims, coord.points, coord_attrs)
+        else:
+            coords[coord.var_name] = ((), np.asscalar(coord.points), coord_attrs)
+
+    array_attrs = _iris_obj_to_attrs(cube)
+    cell_methods = _iris_cell_methods_to_str(cube.cell_methods)
+    if cell_methods:
+        array_attrs['cell_methods'] = cell_methods
+
+    # Deal with iris 1.* and 2.*
+    cube_data = cube.core_data() if hasattr(cube, 'core_data') else cube.data
+
+    # Deal with dask and numpy masked arrays
+    if dask_ma and isinstance(cube_data, Array):
```

Comment:
Also, again it would be better to use `dask_array_type` andd import `dask.array.ma` here rather than above. That would ensure that an error is raised if somehow a cube contains a dask array but has the wrong version of dask installed, rather than letting the error pass silently.